### PR TITLE
Border class and registry

### PR DESCRIPTION
**Pull request by polypoyo**
Makes it possible to split up your custom borders. Tested with `glow` and `battle` borders in Dark Place Legacy.

### CODE REVIEW

**Comment by polypoyo** [comment]
> Tested with `glow` and `battle` borders in Dark Place Legacy.

Should clarify that I mean I tested that those work without modification.

**Comment by FireRainV** [comment]
Make sure to also change the borders debug menu to use the registry instead of manually searching.

**Comment by polypoyo** [comment]
Thanks for bringing that to my attention! For now, it will check both the files and the registry.

**Comment by FireRainV** [comment]
why didn't you remove the old code? you're basically adding the same borders twice.

**Comment by FireRainV** [comment]
nvm, it's intentional

File: mods/_testmod/scripts/borders/spinnysquare.lua [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    )
+    love.graphics.rotate(math.rad(20))
+    love.graphics.rectangle("fill", -1000*BORDER_SCALE, -1000*BORDER_SCALE, 1000*2*BORDER_SCALE, 1000*2*BORDER_SCALE)
+    self:drawRect(max_depth - 1, not white)
+    love.graphics.pop()
+end
+
+function SpinnySquare:draw()
+    love.graphics.push("all")
+    love.graphics.translate(1920*0.5*BORDER_SCALE, 1080*0.5*BORDER_SCALE)
```
Comment (FireRainV) [inline mods/_testmod/scripts/borders/spinnysquare.lua:30]: what does this line do? it won't have issues for monitors that are non-1080p, right? I have a 1440p monitor.

Comment (polypoyo) [reply mods/_testmod/scripts/borders/spinnysquare.lua:30]: This is supposed to move the 0,0 from the top left of the 16:9 to the middle of the screen for the drawRect method. I don't know if it works for 1440p.

Comment (FireRainV) [reply mods/_testmod/scripts/borders/spinnysquare.lua:30]: yea, I changed to this branch and it seems to work fine, but would it look on non-16:9 screens like mobiles and widescreens?:
image

Comment (polypoyo) [reply mods/_testmod/scripts/borders/spinnysquare.lua:30]: On 21:9 it works fine. The `love.graphics.translate` from my ultrawide PR applies here.

**Comment by NyakoFox** [comment]
Does this break backwards compatibility?

**Comment by polypoyo** [comment]
> Does this break backwards compatibility?

AFAIK no. I've tested with Dark Place Legacy's animated borders (KRISTAL_EVENT.onBorderDraw), and those are fine. Also tested with image borders in StarRune

**Pull request merged**